skrifa/src/outline/autohint/style.rs

```diff
@@ -0,0 +1,235 @@
+//! Per-glyph style assignment.
+
+use super::script::{ScriptClass, ScriptRange, SCRIPT_RANGES};
+use crate::GlyphId;
+
+/// Defines the script and style associated with a single glyph.
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+#[repr(transparent)]
+pub(super) struct GlyphStyle(pub(super) u16);
+
+impl GlyphStyle {
+    // The following flags roughly correspond to those defined in FreeType
+    // here: https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/autofit/afglobal.h#L76
+    // but with different values because we intend to store "meta style"
+    // information differently.
+    pub const SCRIPT_INDEX_MASK: u16 = 0xFF;
+    pub const UNASSIGNED: u16 = Self::SCRIPT_INDEX_MASK;
+    pub const NON_BASE: u16 = 0x100;
+    pub const DIGIT: u16 = 0x200;
```

Comment:
are these actually used anywhere outside of this impl block?

Reply:
`NON_BASE` is currently used in `script.rs` and there are other passes that may modify the values. This type is very likely to change as I port more code so I'd prefer to keep it as is for now and do some cleanup after I've covered all potential uses.